src/meili/candidate_search.cc

```diff
@@ -128,12 +130,17 @@ CandidateCollector::WithinSquaredDistance(const midgard::PointLL& location,
 
 std::vector<std::vector<baldr::PathLocation>>
 CandidateQuery::QueryBulk(const std::vector<midgard::PointLL>& locations,
+                          const std::vector<baldr::Location::StopType>& stop_types,
                           float radius,
                           sif::EdgeFilter filter) {
+  if (locations.size() != stop_types.size())
```

Comment:
Would it make more sense to have the arguments to this function be correct by construction? i.e., zipping these into a `pair` or `tuple`?

For example: 

```C++
CandidateQuery::QueryBulk(const std::vector<std::pair<midgard::PointLL, baldr::Location::StopType>& locations)
```

Reply:
zipping wouldnt make sense, but it would make more sense to pass the `Measurement` directly. i considered doing that but if you notice there it takes the radius which it cannot get directly from measurement because its a function of two parts of the measurement. let me see what it would take to move this radius limiting inside of the query function.

Reply:
Why wouldn't it make sense? ... oh are you saying that it would make more sense to refactor this to accept `Measurement`, which includes all the query parameters? Yeah, that's definitely a better alternative.

Reply:
Definitely that yes but also there is no idiomatic way to zip like in python. to do it that way we would suffer many thousands of items copied from one place in memory to the next